src/experimental/mdspan

```diff
@@ -43,22 +43,8 @@
 
 #pragma once
 
-#ifdef _MSC_FULL_VER
-// - 4348: redefinition of default parameter
-#pragma warning(push)
-#pragma warning(disable : 4348)
-#endif
-
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunused-parameter"
-#pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
-#pragma clang diagnostic ignored "-Wreserved-id-macro"
-#endif
```

Comment:
We added all of the above suppressions.  Probably should keep these.  You may need to add a new one for gcc.

```c++
#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Werror=undef"
#endif

[original content of mdspan goes here]

#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic pop
#endif
```

Reply:
Optionally, create a new file `mdspan_wrapper.hh` that contains all of these pragmas and has one line of code: `#include <experimental/mdspan`.  This might might updating mdspan easier.

Reply:
Since a separate wrapper header file will I think require one of (1) switching `#include <mdspan>` (which is nice because it mimics future STL behavior) with `#include <mdspan_wrapper> (2) changing the kokkos/mdspan `mdspan` to e.g. `mdspan_wrapped` and including this in a new `mdspan` file, I'm in favor of just leaving things the way they are now. Also I now know how to re-add the compiler guards (which may anyways change with future versions of mdspan...)